datalad/interface/tests/test_add_archive_content.py

```diff
@@ -192,21 +205,35 @@ def test_add_archive_content(path_orig, url, repo_path):
     with chpwd(repo_path):
         # TODO we need to be able to pass path into add_archive_content
         # We could mock but I mean for the API
-        assert_raises(RuntimeError, add_archive_content, "nonexisting.tar.gz") # no repo yet
 
-        repo = AnnexRepo(repo_path, create=True)
-        assert_raises(ValueError, add_archive_content, "nonexisting.tar.gz")
+        # no repo yet
+        assert_raises(NoDatasetFound, add_archive_content, "nonexisting.tar.gz")
+        ds = Dataset(repo_path).create()
+        res = ds.add_archive_content("nonexisting.tar.gz", on_failure='ignore')
+        assert_in_results(res, action='add-archive-content',
+                          status='impossible')
+
         # we can't add a file from outside the repo ATM
-        assert_raises(FileNotInRepositoryError, add_archive_content, opj(path_orig, '1.tar.gz'))
+        res = ds.add_archive_content(Path(path_orig) / '1.tar.gz',
+                                     on_failure='ignore')
+        assert_in_results(
+            res,
+            action='add-archive-content',
+            status='impossible',
+            type="dataset",
+            message="Can not add archive outside of the dataset"
+              )
+
 
         # Let's add first archive to the repo so we could test
         with swallow_outputs():
-            repo.add_urls([opj(url, '1.tar.gz')], options=["--pathdepth", "-1"])
+            ds.repo.add_urls([opj(url, '1.tar.gz')], options=["--pathdepth", "-1"])
```

Comment:
`.repo` is "expensive" so test could minimize diff and get speedier a bit by binding `repo = ds.repo`

might be relevant to other tests as well

Reply:
should be done in 145d048